### PR TITLE
Modal route switch

### PR DESCRIPTION
**Pull request by kenjinp**
Adds modal switch routing like in dribble or pinterest or twitter, for the `ResourceView`

Screenshot 2020-01-30 at 17 00 49

### CODE REVIEW

File: src/shared/views/ResourceView.tsx [hunk 1]
```diff
@@ -1,367 +1,40 @@
...
-        >
-          {resource && <GraphContainer resource={resource} />}
-        </div>
-      </div>
+      <ResourceViewContainer
+        render={resource => (
+          <div
+            ref={ref}
+            className="graph-wrapper"
+            style={{
```
Comment (dhaneshnm) [inline src/shared/views/ResourceView.tsx]: Nit Pick: Do we need these inline styles here? Can we move them to graph-wrapper to create or create a new class?

File: src/shared/views/ResourceView.tsx [hunk 2]
```diff
@@ -1,367 +1,40 @@
...
+              width: '100%',
+              height: '100%',
+              position: 'relative',
+            }}
+          >
+            <div
+              style={{
```
Comment (dhaneshnm) [inline src/shared/views/ResourceView.tsx]: Earlier comment about inline styles ..

File: src/shared/containers/ResourceViewContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  const handleGoToInternalLink = (link: ResourceLink) => {
+    const { orgLabel, projectLabel } = getOrgAndProjectFromProjectId(
+      (link as IncomingLink)._project
+    );
+    goToResource(orgLabel, projectLabel, encodeURIComponent(link['@id']), {
+      tab: '#links',
+    });
+  };
+
+  useAsyncEffect(async () => {
```
Comment (dhaneshnm) [inline src/shared/containers/ResourceViewContainer.tsx]: Can we use regular useEffect? In the spirit of removing useAsyncEffect from the code base?

File: src/shared/containers/ResourceViewContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+                            });
+                          }}
+                        >
+                          Revision {rev}
+                        </a>
+                      );
+                    }}
+                  />
+                </TabPane>
+                <TabPane tab="Links" key="#links">
```
Comment (dhaneshnm) [inline src/shared/containers/ResourceViewContainer.tsx]: Can we remove inline styling where possible in the below section?

File: src/shared/App.tsx [hunk 5]
```diff
@@ -1,22 +1,56 @@
...
+    <MainLayout>
+      <Switch location={background || location}>
+        {routes.map(({ path, component: C, ...rest }) => (
+          <Route key={path as string} path={path} component={C} {...rest} />
+        ))}
+        <Route component={NotFound} />
+      </Switch>
+
+      {
+        // This is where special routes should go that
```
Comment (Stafeeva) [inline src/shared/App.tsx]: there is probably one extra 'that' in this comment

**Pull request merged**